presto-main/src/main/java/io/prestosql/operator/TableWriterOperator.java

```diff
@@ -243,6 +259,15 @@ public void addInput(Page page)
         updateWrittenBytes();
     }
 
+    private void verifyBlockHasNoNulls(Block block, String columnName)
+    {
+        for (int position = 0; position < block.getPositionCount(); position++) {
```

Comment:
The block check logically fits better here
```java
if (!block.mayHaveNull()) {
    return;
}

Reply:
Done in the latest force push.